File: 2128-reverse-prefix-of-word/2128-reverse-prefix-of-word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function(word, ch) {
    let start = 0, end = 0;
    for(let i=0; i<word.length; i++){
        if(word[i] === ch){
            end = i;
            break;
        }
    }
    word = word.split("");
    while(start < end){
        [word[start], word[end]] = [word[end], word[start]];
        start++; end--;
    }

    return word.join("");
};

File: 2128-reverse-prefix-of-word/2128-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int start = 0, end = 0;
        StringBuilder reverse = new StringBuilder(word);
        for(int i=0; i<word.length(); i++){
            if(word.charAt(i) == ch){
                end = i;
                break;
            }
        }
        while(start<end){
            char temp = reverse.charAt(end);
            reverse.setCharAt(end, reverse.charAt(start));
            reverse.setCharAt(start, temp);
            start++;
            end--;
        }

        return reverse.toString();
    }
}